RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+---
+RFC: RFCXXX-Native-Json-Adapters.md
+Author: Jim Truher
+Status: Draft
+SupercededBy: 
+Version: 0.8
+Area: engine
+Comments Due: March 1, 2023
+Plan to implement: Yes
+---
+
+# Title
+
+Improving the User Experience of Native Applications
+
+## Motivation
+
+As a user I can interact with native applications which can produce a more object like experience.
+
+### Native Applications which Emit Structured Data
+
+Native command execution usually results in textual output which is a fairly poor experience for our users.
+Some of our users use tools like `Crescendo` to allow those tools to "play nice" in the PowerShell arena,
+but the problem of output handling is still troublesome.
+
+Some applications (docker, kubectl, others) can emit in `JSON` which is much easier for our users to convert to useful results in PowerShell,
+but the user still has to know the incantation to have the tool emit `JSON` and then they must pipe that output to `Convert-FromJson`.
+It would be a much better experience if the user could simply type:
+
+```powershell
+PS> docker image ls
+```
+
+rather than
+
+```powershell
+PS> docker image ls --format='{{json .}}' | ConvertFrom-Json | Format-Table
+```
+
+There are two aspects of making this work well:
+
+- Automatically adding `ConvertFrom-Json` to the pipeline following the native command to automatically convert `Json` to object
+- Automatically adding a parameter to a native command similarly to what `$PSDefaultParameterValue` does for cmdlets
+
+### Automatically converting structured text data to PowerShell objects
+
+We should attempt to improve the experience for these users where we can by providing simple way to implicitly support
+extending the pipeline with a JSON to object converter.
+
+Custom output filters can be written to take regular or irregular output and convert it to json or objects.
+The output filter would also need to have the parameters in use for the original native application.
+This would enable a single filter to be used with any combination of parameters.
+If these custom filters could be added automatically to the invocation of the native app,
+the experience for the user would again be greatly improved.
+
+for example:
+
+```powershell
+PS> MyCustomApp.exe -p1 MyString -p2 42 -p3 resource=blue
+```
+
+_and_ if `MyCustomApp-json` is available as a conversion too for `MyCustomApp.exe`,
+we could automatically construct the following:
+
+```powershell
+PS> MyCustomApp.exe -p1 MyString -p2 42 -p3 resource=blue | MyCustomApp-json -ParameterBlock @{ p1 = "MyString"; p2 = 42; p3 = "resource=blue" } | ConvertFrom-Json
```

Comment:
Is there a case where the JSON adapter needs `ConvrtFrom-Json -AsHashTable` as in the case of an empty key or keys that differ only in casing